src/sentry/integrations/vercel/webhook.py

```diff
@@ -267,11 +294,14 @@ def _delete(self, external_id, configuration_id, request):
 
     def _deployment_created(self, external_id, request):
         payload = request.data["payload"]
+        vercel_project_id = (
+            payload["projectId"] if payload.get("projectId") else payload["project"]["id"]
+        )
```

Comment:
Change the way the project id is accessed so that it supports new and old payloads